Source/CombatExtended/CombatExtended/TurretTracker.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using Verse;
+
+namespace CombatExtended
+{
+    [StaticConstructorOnStartup]
+    public class TurretTracker : MapComponent
+    {
+        public List<Building_TurretGunCE> Turrets = new List<Building_TurretGunCE>();
```

Comment:
I'd go for a Set for O(1) insertion/removal (not that it matters in practice if the player has thousands upon thousands of turrets, but still)

Reply:
Done.